pkg/kubelet/dockertools/docker.go

```diff
@@ -316,7 +316,11 @@ func (d *dockerContainerCommandRunner) PortForward(podInfraContainerID string, p
 	}
 
 	containerPid := container.State.Pid
-	// TODO use exec.LookPath for socat / what if the host doesn't have it???
+	// TODO what if the host doesn't have it???
+	path, lookupErr := exec.LookPath("socat")
+	if lookupErr != nil {
+		return fmt.Errorf("The socat command cannot be find in PATH=%s.", path)
```

Comment:
I can't remember if this error is returned to the client, but if so, I don't think it's appropriate to tell the end user what the PATH is on the kubelet. You could certainly log this so it shows up in the kubelet's log file, but I don't think the end user should get the PATH info.